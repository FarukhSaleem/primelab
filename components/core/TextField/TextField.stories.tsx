import React from "react";
import { Box } from "@mui/material";
import { ComponentMeta } from "@storybook/react";
import TextField from "./TextField";

export default {
  title: "Component/TextField",
  component: TextField,
  decorators: [
    (Story) => (
      <Box maxWidth="lg">
        <Story />
      </Box>
    ),
  ],
} as ComponentMeta<typeof TextField>;
export const TextFieldWithLabel = ({
  label = "Test",
  name = "test",
}: {
  label: string;
  name: string;
}) => <TextField label={label} name={name} />;
