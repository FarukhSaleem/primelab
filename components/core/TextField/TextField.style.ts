import {makeStyles} from "@mui/styles";

export const useStyles = makeStyles((theme: any) => ({
  '& .MuiInput-formControl': {
    height: '100%',
    alignItems: 'center',
  },
  textField: {
    marginTop: 13,
    marginBottom: 20, 
  },
}))
export default useStyles;