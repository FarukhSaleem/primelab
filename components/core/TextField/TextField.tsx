import React from "react";
import MaterialTextField from "@mui/material/TextField";
import { Props } from "./TextField.type";
import useStyles from "./TextField.style";

const TextField = ({ label, id, name, ...restProps }: Props): JSX.Element => {
  const classes = useStyles();
  return (
    <MaterialTextField
      className={classes.textField}
      required
      fullWidth
      label={label}
      id={id}
      name={name}
      {...restProps}
    />
  );
};
export default TextField;
