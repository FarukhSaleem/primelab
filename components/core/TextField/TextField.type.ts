import { TextFieldProps } from "@mui/material/TextField";

export type Props = {
  label: string;
} & TextFieldProps;
