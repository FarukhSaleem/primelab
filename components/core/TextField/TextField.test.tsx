import React from "react";
import { fireEvent, render } from "@testing-library/react";
import TextField from "./TextField";
import { Props } from "./TextField.type";

function renderButton(props: Props) {
  return render(<TextField {...props} />);
}

describe("<TextField />", () => {
  test("It should render input field", async () => {
    const result = renderButton({ label: "TextField", className: "MyTextField" });
    const textElement = result.baseElement.getElementsByTagName("input")[0]
    expect(textElement).toMatchSnapshot();
    fireEvent.focus(textElement);
    fireEvent.change(textElement, {target: {value: '23'}})
    expect(textElement.value).toBe('23')
  });
});
