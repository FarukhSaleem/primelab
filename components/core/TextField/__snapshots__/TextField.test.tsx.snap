// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> It should render input field 1`] = `
<input
  aria-invalid="false"
  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
  id="mui-1"
  required=""
  type="text"
  value=""
/>
`;
